lms/static/js/certificates/collections/certificate_whitelist.js

```diff
@@ -0,0 +1,58 @@
+// Backbone.js Application Collection: CertificateWhiteList
+/*global define, RequireJS */
+
+;(function(define){
+    'use strict';
+    define([
+            'backbone',
+            'gettext',
+            'js/certificates/models/certificate_exception'
+        ],
+
+        function(Backbone, gettext, CertificateExceptionModel){
+
+            var CertificateWhiteList =  Backbone.Collection.extend({
+                model: CertificateExceptionModel,
+
+                initialize: function(attrs, options){
+                    this.url = options.url;
+                },
+
+                getModel: function(attrs){
+                    var model = this.findWhere({user_name: attrs.user_name});
+                    if(attrs.user_name && model){
```

Comment:
@mattdrayer in case of empty attrs.user_name 'findWhere' will return a model instead of 'undefined', this case arises if a model is already added with an email and for such a model user_name would be empty.